ir/expression.def

```diff
@@ -141,8 +141,32 @@ class Constant : Literal {
 #noconstructor
     /// if noWarning is true, no warning is emitted
     void handleOverflow(bool noWarning);
+    // We need to enumerate all the integer types because we need proper 64-bit handling on
+    // 32-bit systems (which ain't long!) and mpz_import is too big a hammer because and it loses
+    // the signess of the value.
+    Constant(int v, unsigned base = 10) :
+        Literal(new Type_InfInt()), value(v), base(base) {}
+    Constant(unsigned v, unsigned base = 10) :
+        Literal(new Type_InfInt()), value(v), base(base) {}
+#emit
+#if __WORDSIZE == 64
+    Constant(intmax_t v, unsigned base = 10) :
+        Literal(new Type_InfInt()), value(v), base(base) {}
+#else
+    Constant(long v, unsigned base = 10) :
+        Literal(new Type_InfInt()), value(v), base(base) {}
+    Constant(unsigned long v, unsigned base = 10) :
+        Literal(new Type_InfInt()), value(v), base(base) {}
     Constant(intmax_t v, unsigned base = 10) :
-            Literal(new Type_InfInt()), value((long)v), base(base) {}
+        Literal(new Type_InfInt()), base(base) {
+        mpz_import(value.get_mpz_t(), /* word count */1, /* msb first */0,
+                   /* word size */sizeof(v), /* native endian */0, /* nails */0, &v); }
```

Comment:
I suspect what might be required here is to check the sign of `v` and negate it before calling mpz_import if it is negative (so the value passed to mpz_import is always positive), then negate the mpz_class afterwards if it was originally negative.  With that, you just need two constructors (intmax_t and uintmax_t) -- all other integer types should convert to one of those two, plus you don't need any special-case for WORDSIZE.

Reply:
Maybe true. We did not have an unsigned constructor before, we were simply using a cast to long, and that was causing problems. I suspect your solution could work, and I will give it a try later. I do want to get this in now to unblock the pipeline, and I will revisit.